lib/utils/tar.js

```diff
@@ -136,8 +136,10 @@ BundledPacker.prototype.applyIgnores = function (entry, partial, entryObj) {
 
     // the package root.
     var p = this.parent
-    // the package before this one.
+    // the directory before this one.
     var pp = p && p.parent
+    // the directory before that (if this is scoped)
+    if (pp && pp.basename[0] === '@') pp = pp && pp.parent
```

Comment:
Incidentally, `fstream-npm` doesn't require a matching patch. This patch is required because `this.isBundled` does not entirely mesh with`fstream-npm` as I didn't understand `fstream-npm`'s filtering method very well when I wrote it. (For instance, I didn't know that `applyIgnores` would get called recursively.)